File: java/06_pgm_Avion/src/co/edu/campusucc/poo/TipoComercial.java
package co.edu.campusucc.poo;

public class TipoComercial extends Avion {

    public TipoComercial() {
        super();
        setTipo("Comercial");
    }

    @Override
    public void startingEngine() {
        System.out.println("⛽Check Fuel ✅");
        for (int i = 0; i < 3; i++) {
            System.out.println("💥💥Spark💥💥");
            try {
                Thread.sleep(1000 - (i * 200));
            } catch (InterruptedException e) {
                System.err.println("⛔:" + e);
            }
        }
        System.out.println("Started the Motor...✈️✈️");
    }

    @Override
    public void stopingEngine() {
        System.out.println("✈️✈️ Check Engine ✅");
        for (int i = 0; i < 2; i++) {
            System.out.println("⛔ Stoping Engine...⛔‼️");
            try {
                Thread.sleep(1000 - (i * 400));
            } catch (InterruptedException e) {
                System.err.println("⛔:" + e);
            }
        }
        System.out.println("Stoped...⛔");
    }

}

File: c++/06_pgm_avion/src/co/edu/campusucc/poo/TipoComercial.cpp
#include <iostream>
#include <string>
#include "Avion.h"

class TipoComercial : public Avion {
public:
    TipoComercial() {
        Avion();
        setTipo("Comercial");
    }

    void startingEngine() override {
        std::cout << "⛽Check Fuel ✅" << std::endl;
        for (int i = 0; i < 3; i++) {
            std::cout << "💥💥Spark💥💥" << std::endl;
            try {
                std::this_thread::sleep_for(std::chrono::milliseconds(1000 - (i * 200)));
            } catch (std::exception &e) {
                std::cerr << "⛔:" << e.what() << std::endl;
            }
        }
        std::cout << "Started the Motor...✈️✈️" << std::endl;
    }

    void stopingEngine() override {
        std::cout << "✈️✈️ Check Engine ✅" << std::endl;
        for (int i = 0; i < 2; i++) {
            std::cout << "⛔ Stoping Engine...⛔‼️" << std::endl;
            try {
                std::this_thread::sleep_for(std::chrono::milliseconds(1000 - (i * 400)));
            } catch (std::exception &e) {
                std::cerr << "⛔:" << e.what() << std::endl;
            }
        }
        std::cout << "Stoped...⛔" << std::endl;
    }
};
